notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "  \"IATA_CODE_Reporting_Airline\", \"Tail_Number\", \"Flight_Number_Reporting_Airline\", \n",
+    "  \"OriginAirportID\", \"OriginAirportSeqID\", \"OriginCityMarketID\", \"Origin\", \"OriginCityName\", \"OriginState\", \"OriginStateFips\", \"OriginStateName\", \"OriginWac\",  \n",
+    "  \"DestAirportID\", \"DestAirportSeqID\", \"DestCityMarketID\", \"Dest\", \"DestCityName\", \"DestState\", \"DestStateFips\", \"DestStateName\", \"DestWac\", \n",
+    "  \"CRSDepTime\", \"DepTime\", \"DepDelay\", \"DepDelayMinutes\", \"DepDel15\", \"DepartureDelayGroups\", \"DepTimeBlk\",\n",
+    "  \"TaxiOut\", \"WheelsOff\", \"WheelsOn\", \"TaxiIn\", \"CRSArrTime\", \"ArrTime\", \"ArrDelay\", \"ArrDelayMinutes\", \"ArrDel15\", \"ArrivalDelayGroups\", \"ArrTimeBlk\",\n",
+    "  \"Cancelled\", \"CancellationCode\", \"Diverted\", \"CRSElapsedTime\", \"ActualElapsedTime\", \"AirTime\", \"Flights\", \"Distance\", \"DistanceGroup\", \"CarrierDelay\",\n",
+    "  \"WeatherDelay\", \"NASDelay\", \"SecurityDelay\", \"LateAircraftDelay\", \"FirstDepTime\", \"TotalAddGTime\", \"LongestAddGTime\", \"DivAirportLandings\", \"DivReachedDest\",\n",
+    "  \"DivActualElapsedTime\",  \"DivArrDelay\",  \"DivDistance\",  \"Div1Airport\",  \"Div1AirportID\",  \"Div1AirportSeqID\",  \"Div1WheelsOn\",  \"Div1TotalGTime\",\n",
+    "  \"Div1LongestGTime\",  \"Div1WheelsOff\",  \"Div1TailNum\",  \"Div2Airport\",  \"Div2AirportID\",  \"Div2AirportSeqID\",  \"Div2WheelsOn\",  \"Div2TotalGTime\",\n",
+    "  \"Div2LongestGTime\",  \"Div2WheelsOff\",  \"Div2TailNum\",  \"Div3Airport\",  \"Div3AirportID\",  \"Div3AirportSeqID\",  \"Div3WheelsOn\",  \"Div3TotalGTime\",\n",
+    "  \"Div3LongestGTime\",  \"Div3WheelsOff\",  \"Div3TailNum\",  \"Div4Airport\",  \"Div4AirportID\",  \"Div4AirportSeqID\",  \"Div4WheelsOn\",  \"Div4TotalGTime\",\n",
+    "  \"Div4LongestGTime\",  \"Div4WheelsOff\",  \"Div4TailNum\",  \"Div5Airport\",  \"Div5AirportID\",  \"Div5AirportSeqID\",  \"Div5WheelsOn\",  \"Div5TotalGTime\",\n",
+    "  \"Div5LongestGTime\",  \"Div5WheelsOff\",  \"Div5TailNum\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY YEAR'''\n",
+    "\n",
+    "display.run_task(sql)\n",
+    "sql_client.wait_until_ready('flights_year')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "36f0ddff-8a25-4c47-afe3-fec08066e11a",
+   "metadata": {},
+   "source": [
+    "Here's the link to view the segments for this one on [the Apache Druid console in the Segments view](http://localhost:8888/unified-console.html#segments/datasource~flights_year).\n",
+    "\n",
+    "Notice that now we have all the rows in a single segment and that it has over 500,000 rows. Normally this would still be a small segment, but our target for this discussion is 50,000. This solution puts us at 10x.\n",
+    "![](assets/segments-yearly.png)\n",
+    "\n",
+    "\n",
+    "### Almost there...\n",
```

Comment:
This is a great example of using a non-text granularity. I'd focus on that over targets on row counts just yet. However, I note that you're PARTITIONED BY YEAR?